src/site/content/en/blog/generic-sensor/index.md

```diff
@@ -0,0 +1,545 @@
+
+{% Aside %} As of February 2021, GravitySensor is still
+[under development](https://bugs.chromium.org/p/chromium/issues/detail?id=1163993) with no estimated
+release date. {% endAside %}
+
+### Gyroscope {: #gyroscope-sensor }
+
+ <figure class="w-figure">
+  <video controls autoplay loop muted class="w-screenshot">
+    <source src="gyroscope.mp4" type="video/mp4">
+  </video>
+  <figcaption class="w-figcaption">
+    Gyroscope sensor measurements
+  </figcaption>
+</figure>
+
+The [`Gyroscope`](https://developer.mozilla.org/en-US/docs/Web/API/Gyroscope) sensor measures
+angular velocity in radians per second around the device's local X, Y, and Z axis. Most consumer
+devices have mechanical ([MEMS](https://en.wikipedia.org/wiki/Microelectromechanical_systems))
+gyroscopes, which are inertial sensors that measure rotation rate based on
+[inertial Coriolis force](https://en.wikipedia.org/wiki/Coriolis_force). MEMS gyroscopes are prone
+to drift that is caused by the sensor's gravitational sensitivity which deforms the sensor's
+internal mechanical system. Gyroscopes oscillate at relative high frequencies, e.g., 10s of kHz, and
+therefore, might consume more power compared to other sensors.
+
+### Orientation sensors {: #orientation-sensors }
+
+<figure class="w-figure">
+  <video controls autoplay loop muted class="w-screenshot">
+    <source src="orientation.mp4" type="video/mp4">
+  </video>
+  <figcaption class="w-figcaption">
+    Absolute orientation sensor measurements
+  </figcaption>
+</figure>
+
+The
+[`AbsoluteOrientationSensor`](https://developer.mozilla.org/en-US/docs/Web/API/AbsoluteOrientationSensor)
+is a fusion sensor that measures rotation of a device in relation to the Earth's coordinate system,
+while the
+[`RelativeOrientationSensor`](https://developer.mozilla.org/en-US/docs/Web/API/RelativeOrientationSensor)
+provides data representing rotation of a device hosting motion sensors in relation to a stationary
+reference coordinate system.
+
+All modern 3D JavaScript frameworks support [quaternions](https://en.wikipedia.org/wiki/Quaternion)
+and [rotation matrices](https://en.wikipedia.org/wiki/Rotation_matrix) to represent rotation;
+however, if you use WebGL directly, the `OrientationSensor` conveniently has both a
+[`quaternion` property](https://developer.mozilla.org/en-US/docs/Web/API/OrientationSensor/quaternion)
+and a
+[`populateMatrix()` method](https://developer.mozilla.org/en-US/docs/Web/API/OrientationSensor/populateMatrix).
+Here are few snippets:
+
+**[three.js](https://threejs.org/docs/index.html#api/core/Object3D.quaternion)**
+
+```js
+let torusGeometry = new THREE.TorusGeometry(7, 1.6, 4, 3, 6.3);
+let material = new THREE.MeshBasicMaterial({ color: 0x0071c5 });
+let torus = new THREE.Mesh(torusGeometry, material);
+scene.add(torus);
+
+// Update mesh rotation using quaternion.
+const sensorAbs = new AbsoluteOrientationSensor();
+sensorAbs.onreading = () => torus.quaternion.fromArray(sensorAbs.quaternion);
```

Comment:
We should be consistent in this article and pick either `onreading` or `addEventListener('reading', ...`.
It applies to `onerror` as well.

Reply:
`onreading` and `onerror` are the winners!